sqlobject/src/main/java/org/jdbi/v3/sqlobject/locator/SqlLocator.java

```diff
@@ -29,6 +29,21 @@ public interface SqlLocator {
      * @param method the method
      * @param config the config registry
      * @return the SQL for the given method.
+     * @deprecated use {@link #locate(Class, Object, Method, ConfigRegistry)}
      */
+    @Deprecated
     String locate(Class<?> sqlObjectType, Method method, ConfigRegistry config);
+
+    /**
+     * Locates and returns the SQL for the given SQL Object type, target and method.
+     *
+     * @param sqlObjectType the SQL object type
+     * @param target the target
+     * @param method the method
+     * @param config the config registry
+     * @return the SQL for the given method.
```

Comment:
I understand that this is the javadoc copied from the other method; this is a good opportunity to improve it. To the uninitiated, it is not clear what "the method" (from the SQL object type) is or what "the target" (which is the new addition) is. Splitting the sql object type and the method does not help.